### PR TITLE
feat: start tracing request with opentelemetry

### LINKED ISSUE #1031: FR(gcf-utils): enable cloud trace for bots
It would be nice to add additional observability for our GCF bots.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1031

### CODE REVIEW

File: packages/gcf-utils/src/gcf-utils.ts [hunk 1]
```diff
@@ -307,25 +336,35 @@ export class GCFBootstrapper {
...
+          const span = tracer.startSpan('enqueue task');
+          try {
+            await this.enqueueTask({
+              id,
+              name,
+              signature,
+              body: JSON.stringify(request.body),
+            });
+          } finally {
+            span.end();
```
Comment (bcoe) [inline packages/gcf-utils/src/gcf-utils.ts:366]: is `span.end` synchronous? wondering if there's a way to await all spans flushing?

Comment (chingor13) [reply packages/gcf-utils/src/gcf-utils.ts:366]: This just marks the span as ended (writes the end timestamp). Completed spans are published when the publisher is force flushed.

Comment (chingor13) [reply packages/gcf-utils/src/gcf-utils.ts:366]: I'm not a fan of these try/finally blocks - most other OpenCensus/OpenTelemetry implementations use callable closures to ensure the spans are closed - but I see how it's an issue with the `async` functions.

**Comment by bcoe** [comment]
If you want to experiment with this, I'm supportive.

**Pull request closed without merging**